calculator/main.py

```diff
@@ -4,30 +4,39 @@
 from calculator.exceptions import *
 
 
-def create_new_calculator(operations=None):
-    """
-    Creates a configuration dict for a new calculator. Optionally pre loads an
-    initial set of operations. By default a calculator with no operations
-    is created. Returns a dict containing operations(dict) and history(list).
-
-    :param operations: Dict with initial operations.
-                       ie: {'sum': sum_function, ...}
-    """
-    pass
-
+def create_new_calculator(operations={}):
+    dictionary = {
+            'operations': operations,
+            'history': []
+        }
+        
+    return dictionary
 
 def perform_operation(calc, operation, params):
     """
+    
     Executes given operation with given params. It returns the result of the
     operation execution.
 
     :param calc: A calculator.
     :param operation: String with the operation name. ie: 'add'
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
+                   
+    def test_perform_operation(self):
+        res = perform_operation(self.calc, 'add', (5, 3))
+        self.assertEqual(res, 8)
     """
-    pass
-
+    for param in params:
```

Comment:
Cool, I didn't think of doing the typechecking in that way (`if type in ...`)
Though, in this case, strictly speaking, you might want to do it another way, since technically you probably would want to allow subtypes of float or int to be allowed if such were around.